### ISSUE TITLE
Cannot install App Center Profile on iPhone XS (iOS 12.3.1)

### ISSUE THREAD

**Issue by ryanmvretail** [microsoft/appcenter#615]
**What App Center service does this affect?**
Device Registration

**Describe the bug**
I logged into App Center in Safari on my iPhone XS running iOS 12.3.1.  I clicked the "Add new device" link which then prompted me to download the configuration profile.  I click "Allow" and get prompted to review the profile in the Settings app.  I go to Settings, Profile Downloaded, click "Install", enter my passcode, click "Install" again.  It loads for a moment then takes me back to the website where it shows my phone added as a device, however if I go back to Settings, General, Profile it doesn't show the profile as Installed, only Downloaded.  If I click the Profile I get prompted to install it again.  This loop in process continues indefinitely, never actually installing the profile, yet never giving me an error.  The website shows my phone as registered along with the UDID, however I can't install any of the applications.  I've tried removing the phone from my registered devices and readding it.  I've also tried removing the downloaded profile and reinstalling.  Nothing resolves the issue.

**To Reproduce**
1. Log into App Center
2. Add new device
3. Click "Allow"
4. Go to Settings, Profile Downloaded
5. Click "Install"
6. Enter PIN
7. Click "Install"
8. Go back to Settings, General, Profile
9. See Profile still in Downloaded status
10. Go to 5

**Expected behavior**
I expect the profile to install successfully and allow my device to install App Center builds.

**Screenshots**
![IMG_0547](https://user-images.githubusercontent.com/2294073/59951652-33895700-943f-11e9-87e6-9804804eba4d.jpg)
![IMG_0548](https://user-images.githubusercontent.com/2294073/59951665-3dab5580-943f-11e9-993c-a1fcf331c173.PNG)
![IMG_0549](https://user-images.githubusercontent.com/2294073/59951671-413edc80-943f-11e9-8b29-79178fb7fe37.jpg)
![IMG_0546](https://user-images.githubusercontent.com/2294073/59951677-4439cd00-943f-11e9-9fb0-f9cb0cfae21c.jpg)

**Smartphone (please complete the following information):**
 - Device: iPhone XS Edit: Reproducible on all iOS devices
 - OS: iOS 12.3.1.
 - Browser Safari

**Comment by lumaxis** [comment]
@ryanmvretail Thanks for reporting this!
Were you by any chance able to very this on any other device? I'm pretty sure the issue is not specific to the XS but would be nice to double check.

**Comment by kjng** [comment]
I'm experiencing the same issue with on iPhone 6S 12.3.1 and iPhone 7 Plus 10.3.3. The profile doesn't seem to actually install.

Releases that were published starting from June 17 to present time won't install for me. I think this might have something to do with the [downtime on iOS signing on June 20](https://status.appcenter.ms/).

**Comment by kjng** [comment]
Testing today, I'm able to install debug builds and the profile appears to be working correctly.

**Comment by dimazaid** [comment]
Thanks for confirming @kjng. @ryanmvretail can you confirm whether this issue is till happening?

**Comment by vtsenev** [comment]
I'm having the exact same issue now on two iPhone XS devices with iOS 12.4.

**Comment by lumaxis** [comment]
Reopening this since it seems to still be occurring.
To clarify: The actual device registration does work, App Center gets the device UDID after redirection to the browser but the installation modal in the iOS settings app is never finished and a user has to exit it via the "Cancel" button.

**Comment by lumaxis** [comment]
Thanks, @no-response 🤦🏻‍♂️

**Comment by zyead** [comment]
Reopened #615.

الحصول على Outlook لـ iOS<https://aka.ms/o0ukef>
________________________________
من: Lukas Spieß <notifications@github.com>
‏‏تم الإرسال: Saturday, August 3, 2019 11:17:22 AM
إلى: microsoft/appcenter <appcenter@noreply.github.com>
نسخة: Subscribed <subscribed@noreply.github.com>
‏‏الموضوع: Re: [microsoft/appcenter] Cannot install App Center Profile on iPhone XS (iOS 12.3.1) (#615)


Reopened #615<https://github.com/microsoft/appcenter/issues/615>.

**Comment by reality24** [comment]
I work for Logitech, Inc, and I have the exact problem since we transitioned from HockeApp to 
Appcenter. I have an iPhone XR running iOS 12.4

This needs to be fixed immediately because I can’t install any new builds for any of our apps. And I need to test.

**Comment by reality24** [comment]
@lumaxis So we believe the issue is has to do with browser/site cookies. Installation is successful every time when installing the App Center Profile using the iOS Safari browser in "Private" mode. 

I hope this helps.

**Comment by dctoai** [comment]
> @lumaxis So we believe the issue is has to do with browser/site cookies. Installation is successful every time when installing the App Center Profile using the iOS Safari browser in "Private" mode.
> 
> We have not tried Chrome Incognito mode though.
> 
> I hope this helps.

I tried in Private mode but it does not help, my application is grayed out when installation is finished and it cannot be opened.

**Comment by lumaxis** [comment]
@reality24 @dctoai This is Issue is explicitly about the App Center configuration profile itself. All issues with installing specific apps are likely separate and unrelated, I recommend to open a support request with us to allow us to look at your problem individually.

**Comment by lassecapel** [comment]
I am facing the same issue as described above. Iphone X (12.4)

**Comment by marcos-calis** [comment]
Same issue here. with 5 devices: iPhone 6, iPhone 6s, Ipad Pro, iPhone XR, Ipad mini

**Comment by reality24** [comment]
@lumaxis This issue isn't about a specific build. It's about the installation of the "App Center Device Registration Service" process and configuration profile. The App Center Profile won't install on my device unless I'm in private browsing mode. And in some cases here at Logitech Inc., even when in private mode, the profile won't install. It just throws us in a continuous loop for installing the profile.  Because of this, many of our users cannot access any of the apps until their device is first registered through App Center. As soon as the profile is installed successfully, then everything works fine.  

Please help resolve this issue ASAP.  We transitioned from HockeyApp because we thought AppCenter was ready, but App Center clearly has issues preventing us from testing successfully.

**Comment by reality24** [comment]
@lumaxis We've tested this on 4 different iOS devices in our office, and these devices have the same problems.

**Comment by turanuk** [comment]
Hey folks we apologize for the inconvenience here.

So we apologize for the inconvenience here. For the time being however, even though the device profile doesn't install, the UDIDs should still be in App Center and from an end to end perspective you should not be blocked. For the actual application you are attempting to distribute, as long as users have started the device registration flow and the device shows up under "devices" in install.appcenter.ms, that means you can extract their UDID and include it in your build, either manually, or via our auto-provisioning feature.

The fact that the device registration flow doesn't finish is confusing and we are looking into it in parallel, but providing this information here to ensure that folks aren't blocked in the meantime.

**Comment by reality24** [comment]
@turanuk I don't think you understand. We have not changed our configuration from HockeyApp. Since migration, we aren't able to see apps to download unless our devices are added to App Center.  This process never completes for some phones (same phones we used with hockey app). So it should not be a problem with UDID because 1.) We're using the same devices we have been using in Hockey. 2.) We are using enterprise profile for internal app testing; same as when we were using Hockey.

**Comment by turanuk** [comment]
Right I understand that the process does not complete on App Center which is the issue we are investigating right now. I think what I'm trying to say is that although the process does not complete, you should be re-directed to the App Center website where the device shows up and at which point you should be able to extract the UDID for inclusion in your provisioning profile. Are you seeing that part of the process also fail?

**Comment by reality24** [comment]
@turanuk No that part passes. But why do I need to extract the UUIDs?  This device has already been added in HockeyApp and we had no problems previously. We've been testing with these same devices for ever. Only when we move to AppCenter we have this problem.

**Comment by turanuk** [comment]
Sorry I had missed that you were using an Enterprise provisioning profile, in this case there isn't a difference, but also it means that the App Center device registration flow definitely isn't needed. Can you open a support case where we can get more information about your app and the profile being used?

**Comment by reality24** [comment]
@turanuk  Is there a customer support number, or can we hop on a video call to show you exactly what we are experiencing here at Logitech?

**Comment by turanuk** [comment]
Yeah so please follow this process, which will open a ticket where we can gather more information about your app and investigate further.

> Alternatively, click on the Contact us option in the question menu on the top right of the portal page.

https://docs.microsoft.com/en-us/appcenter/general/support-center#contact-us.

**Comment by dbquarrel** [comment]
You guys need to have an easier and more straight forward path for Hockey migrating people to install these configuration profiles. Apple has made it harder for security purposes to install these. But it's non-obvious to non-technical testers how to install these. Right now if you are migrating from Hockey you have a Hockey configuration profile and to get the new one, if you go to your user page it shows your device registered, but you won't have the profile until you manually install it. So it's counter-intuitive, someone has to realize they don't have a profile (they won't even know what it is), their device looks registered (which it is but it doesn't have the profile) and they need to click the link to register it again (which they won't because the page says it's already registered) in order to download the profile.

So you need to move that out of this page and onto the app page or somewhere simple and clear that says DOWNLOAD CONFIGURATION PROFILE so people can download the thing and we can give them easy instructions to do so. It's buried under too many tiny buttons, misleading links, and incorrect information.

This is for migrating users. Very important.

**Comment by turanuk** [comment]
Thanks @dbquarrel I am cc'ing the Daniel who is the PM for our HockeyApp transition to pass along this feedback cc @derpixeldan 

At a high level, devices are actually not ported from HockeyApp on transition, so I'm surprised that you go to App Center and see your device registered in the system already. We are planning work to merge/improve the install portal so this flow could potentially be addressed there as well. cc @botatoes who is leading that effort.

**Comment by clpolet** [comment]
Quick update on the broken device registration experience. We've found the root cause and are actively working on a fix. 

Keep in mind that this only impacts the user experience while installing the configuration profile. The device registration still works even though the installation of the configuration profile is not completed. App Center gets the device UDID required to register the device at the time you see the App Center portal during the profile installation flow. More information available on [the App Center device registration and app re-signing documentation](https://docs.microsoft.com/appcenter/distribution/auto-provisioning).

Just as a reminder: the configuration profile used in the device registration flow is *not* to be confused with the provisioning profile required by your application for distribution.

**Comment by dwasyluk** [comment]
Impacted by this as well. Experiencing it on iOS 12.3.1/iPhone6 but not on iOS 12.4/iPhone X. Blocks installing on the iPhone6. Says developers are still working on a version for this device even after profile install and appcenter.ms saying device registered.

**Comment by derpixeldan** [comment]
@dbquarrel thanks for the feedback. Installing the profile from App Center is a one time process and solely to register the devices UDID on App Center conveniently. The actual process of installing a profile is rather straight forward and it will get better with iOS 13 where a table row in the first level of the settings pops up once a profile was downloaded. Once the UDID is registered there's nothing to do anymore, so I'm wondering what the issue is you're referring to? This ticket is outlining an issue where the profile is stuck in an infinite loop but it's still registering the UDID, which is what it's supposed to do. As @clpolet pointed out, we're fixing this.

In both #615 and #866 a lot of things are mixed up. What I hear though is that from the following steps, it's unclear what is happening when adding a device and what the install profile is actually doing.

![IMG_3429](https://user-images.githubusercontent.com/3369400/63417612-e5261380-c401-11e9-8de3-0accbebfe114.png)
![IMG_3430](https://user-images.githubusercontent.com/3369400/63417609-e5261380-c401-11e9-8bab-15853b37ef51.png)

**Comment by botatoes** [comment]
Hi there, we have investigated this issue and found that while the user experience for this is terrible, not installing the profile will not block you from any App Center flows. App Center will register your UDID when you hit install the first time. 

We are changing the tester experience and getting rid of the install portal. This issue should be resolved with that effort. [#970]

**Comment by cdub50** [comment]
We are blocked on testing because of this issue cert gets stuck in loop and clicking install on the cert just takes you back to the login page and logging in does nothing to complete the cert install when is this going to be fixed?????

**Comment by mwkoehler** [comment]
I’m happy to see there are eyes on this, but though my phone shows up in my app center account, the installed app remains greyed out no matter how many times I delete it and reinstall.

**Comment by mertnesvat** [comment]
I still experience the not installing Profile issue from the App Center with devices below,
IOS 13.2 iPhone 7 plus
IOS 12.2 iPhone 7

**Comment by mertnesvat** [comment]
https://photos.google.com/share/AF1QipOMCq3IxXONboQwnGFL3ufu0j9o34qQ47jBp2Rox9Gec0o7t6DuIQBjR-trzrXglA/photo/AF1QipNW6LEz7lFrXESmXUL3u4TkHGufHWG-0bPteMUL?key=Uzk4VHNSYi0wUkRrTnN2Rjd5UF9kT09COWJsbDV3 here's a video about couldn't installed configuration profile

**Comment by Amaro96** [comment]
Threre is any solution for that? I´m getting the same issues

**Comment by lumaxis** [comment]
As outlined in [@botatoes' comment](https://github.com/microsoft/appcenter/issues/615#issuecomment-528934384), note that this looks broken right now but the UUID is still correctly registered in App Center.

**Comment by lumaxis** [comment]
That said, we're still working to solve this but need to figure out some unexpected technical issues that are making this take a longer time than anticipated.

**Comment by corodent** [comment]
Our company is also blocked by this. No testers can install right now.

**Comment by mertnesvat** [comment]
Our solution was sadly switching to the Firebase App Distribution. 😞

**Comment by lumaxis** [comment]
@corodent @mertnesvat Can you please elaborate how this is blocking you from installing an app?
The issue described here is simply about the configuration profile screen not dismissing in iOS Settings. The device UDID is still correctly submitted to App Center though and the device is registered to the user. There should be no effect at all on the actual app installation process.

**Comment by lumaxis** [comment]
Locking this issue for now as there seems to be confusion about what the core of the problem is.
This problem does **not** affect functionality and is purely a cosmetic/user experience issue where the device registration flow seems broken but still works.

If you are seeing actual problems with installing issues, please open a new issue or preferably [reach out to our customer support](https://docs.microsoft.com/en-us/appcenter/help#accessing-app-center-free-support).
